test/IoC.Test/UnitTest/IoCTest.cs

```diff
@@ -0,0 +1,42 @@
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IoC.Test.UnitTest;
+
+public class IoCTest
+{
+    private readonly InstancePool _sut; //SUT stands for System Under Test.
+
+    public IoCTest(ITestOutputHelper output)
+    {
+        _sut = InstancePool.Instance;
+    }
+
+    [Fact]
+    public void ShouldGetARegisteredInstance()
+    {
+        //Given
+        _sut.Register<TestClass>();
+
+        //When
+        var tc = _sut.GetInstance<TestClass>();
```

Comment:
Can we also add test that registers multiple instances of the same and confirm memory reference is the same always?

Reply:
Good suggestion. I'll add one test for that.